lib/graphql/batch/setup.rb

```diff
@@ -12,3 +12,12 @@ def after_query(query)
     end
   end
 end
+
+GraphQL::Schema.accepts_definitions(
+  use_graphql_batch: ->(schema) {
+    return if schema.metadata[:graphql_batch_setup]
+    schema.instrument(:query, GraphQL::Batch::Setup)
+    schema.lazy_methods.set(Promise, :sync)
```

Comment:
If you're interested in exploring this option, I'd be happy to add `Schema#lazy_resolve(result_class, method_name)` for this case!